### PR TITLE
feat: Add Config Store support

### PR DESCRIPTION
**Pull request by awilliams-fastly**
- https://developer.fastly.com/reference/api/services/resources/config-store/
- https://developer.fastly.com/reference/api/services/resources/config-store-item/

Related `cli` change: https://github.com/fastly/cli/pull/829

### CODE REVIEW

File: fastly/config_store_item_test.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+			StoreID: cs.ID,
+			Key:     t.Name(),
+			Value:   value,
+		})
+	})
+	if err != nil {
+		t.Fatalf("error creating config store item: %v", err)
+	}
+
+	if got, want := item.StoreID, cs.ID; got != want {
```
Comment (Integralist) [inline fastly/config_store_item_test.go:30]: Just for my own understanding, is it more common to use `have, want` or `got, want` I feel like the former is better due to both words having the same length although `got` is shorter (and that's usually preferred). 

This is more 'bike shedding' than anything so not overly important but from a 'style' perspective I was interested in what we should use for consistency.

**Review by Integralist — COMMENTED** [review]
LGTM. Just a couple of comment.

File: fastly/config_store_item_test.go (hunk 1 above)
Comment (awilliams-fastly) [reply fastly/config_store_item_test.go:30]: I don't think this is a particularly strong convention, but `got/want` is mentioned in the [TestComments](https://github.com/golang/go/wiki/TestComments#got-before-want) Go wiki and is what I see most often.

File: fastly/config_store.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+// CreateConfigStoreInput is the input to CreateConfigStore.
+type CreateConfigStoreInput struct {
+	// Name is the name of the store to create (required).
+	Name string `url:"name"`
+}
+
+// CreateConfigStore creates a new Fastly config store.
+func (c *Client) CreateConfigStore(i *CreateConfigStoreInput) (*ConfigStore, error) {
+	if i.Name == "" {
```
[the remark this answers is not in the document]
Comment (awilliams-fastly) [reply fastly/config_store.go]: Thanks for the context!

Updated in e6d490a to only validate URL path arguments.

File: fastly/config_store_item_test.go (hunk 1 above)
Comment (Integralist) [reply fastly/config_store_item_test.go:30]: Good enough for me 🙂 👍🏻

**Review by Integralist — APPROVED** [review]

**Pull request merged**